src/_Derivatives/linearity.jl

```diff
@@ -38,8 +38,10 @@ function classify_linearity(nd::Vector{NodeData},adj,subexpression_linearity)
             if any_nonlinear
                 # if any children are nonlinear, then we're nonlinear
                 linearity[k] = NONLINEAR
-                # except in the case of ifelse. if the operands are linear then we're piecewise linear
-                if nod.nodetype == CALL && operators[nod.index] == :ifelse
+                # except in the case of ifelse. If the operands are linear then
+                # we're piecewise linear. `nod.index` may be larger than
+                # `operators`, so we use `get`.
+                if nod.nodetype == CALL && get(operators, nod.index, :not_ifelse) == :ifelse
```

Comment:
It would be more consistent with the code below to do something like:
``nod.index < USER_OPERATOR_ID_START && operators[nod.index] == :ifelse``